codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoModGenerator.java

```diff
@@ -63,25 +60,14 @@ static void writeGoMod(
         manifest.addFile(goModFile);
         CodegenUtils.runCommand("go mod init " + settings.getModuleName(), manifest.getBaseDir());
 
-        Map<String, String> externalDependencies = getExternalDependencies(dependencies);
-        for (Map.Entry<String, String> dependency : externalDependencies.entrySet()) {
+        for (Map.Entry<String, String> dependency : goModuleInfo.getMinimumNonStdLibDependencies().entrySet()) {
             CodegenUtils.runCommand(
                     String.format("go mod edit -require=%s@%s", dependency.getKey(), dependency.getValue()),
                     manifest.getBaseDir());
         }
 
         CodegenUtils.runCommand(
-            String.format("go mod edit -go=%s", settings.getGoDirective()),
+            String.format("go mod edit -go=%s", goModuleInfo.getGoDirective()),
             manifest.getBaseDir());
```

Comment:
Not directly related to the changes in the PR, but should we run `go mod tidy` to generate a `go.sum` file?

In [the GitHub Go Modules FAQ](https://github.com/golang/go/wiki/Modules#should-i-commit-my-gosum-file-as-well-as-my-gomod-file), the guidance is to check in the corresponding `go.sum` file produced by `go mod tidy` into version control.